### PR TITLE
Ports guardian rework from hippie

### PR DESCRIPTION
**Pull request by SomeguyManperson**
originalest pr: HippieStation/HippieStation#11848
:cl:  steamp0rt
rscadd: Guardians/holoparasites are no longer seperated into types
rscadd: Instead, you get to "build" one, with a limited amount of points, selecting different stats and abilities
rscadd: Will probably be very unbalanced the first few times.
rscadd: adds the stand arrow, a risky way to draw out potential
rscdel: Goodbye lightning holopara, nobody will ever miss you.
/:cl:

### CODE REVIEW

**Comment by gilvian** [comment]
why more weeb shit

**Comment by nesisknesis** [comment]
fuck you gibong jojo is a good series

**Comment by ynot01** [comment]
HNGGGG

**Comment by SomeguyManperson** [comment]
holyparasite is slightly buffed to heal 0.5 more damage and take 10% less damage oh my god it's OP as fuck

**Comment by MegaEmpirical** [comment]
Never buff mining.

**Comment by LoliconSlayer** [comment]
> Buff Mining
How about no?

**Comment by SomeguyManperson** [comment]
I was assuming that would be the response which is why I kept the arrow ruin out

**Comment by SomeguyManperson** [comment]
added support for randomized guardians and changes the lavaland deck to a random deck

**Comment by boodaliboo** [comment]
nice

**Comment by ynot01** [comment]
look at me Cuccidati, think about who is truly worthy of the arrow!

**Comment by SomeguyManperson** [comment]
Added the gang boss kit to the syndie kit special drop pool, containing a special syndicate arrow that can't be used for requiem, but is laced to work as a mindslave implant

**Comment by MegaEmpirical** [comment]
What the hell does that even mean?

**Comment by SomeguyManperson** [comment]
>there is a new syndie-kit special, which are a set of traitor kits from which one is picked randomly at the cost of 20 TC

>this kit includes a stand arrow which cannot be used to create requiem stands

>this arrow will act as a mindslave implant on people it is used on (arrow have a use limit of 3 by default) This obeys any restrictions already in place for mindslave implants.

File: yogstation/code/modules/guardian/abilities/special/pocket.dm [hunk 1]
```diff
@@ -0,0 +1,387 @@
+GLOBAL_VAR_INIT(pocket_dim, 1)
+GLOBAL_LIST_EMPTY(pocket_mirrors)
+
+/proc/get_final_z(atom/A)
+	var/turf/T = get_turf(A)
+	if(T)
+		return T.z
+	return A.z
+
+/area/hippie/pocket_dimension
```
Comment (monster860) [inline yogstation/code/modules/guardian/abilities/special/pocket.dm]: wow great port keeping the word "hippie" lul

**Review by monster860 — CHANGES_REQUESTED** [review]

File: yogstation/code/modules/guardian/abilities/special/pocket.dm (hunk 1 above)
Comment (SomeguyManperson) [reply yogstation/code/modules/guardian/abilities/special/pocket.dm]: ok fine

Comment (SomeguyManperson) [reply yogstation/code/modules/guardian/abilities/special/pocket.dm]: ez done

**Comment by gilvian** [comment]
JoeJoe still b a d

File: yogstation/code/modules/guardian/abilities/special/pocket.dm [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+
+/datum/guardian_ability/major/special/pocket/proc/check_if_teleport(mob/living/L)
+	var/list/pocket_z = get_pocket_z()
+	if(!pocket_z)
+		return
+	if(get_final_z(L) != pocket_z)
+		take_effects(L)
+
+/datum/guardian_ability/major/special/pocket/proc/add_effects(mob/living/L)
+	L.status_flags |= GODMODE
```
Comment (ghost) [inline yogstation/code/modules/guardian/abilities/special/pocket.dm:117]: Did you port my change that allows you to actually heal in godmode?

File: tgui/src/components/button.ract [hunk 3]
```diff
@@ -5,7 +5,7 @@
   component.exports = {
     computed: {
       clickable () {
-        if (this.get('enabled') && (!this.get('state') || this.get('state') == "toggle")) {
+        if (this.get('enabled') && (!this.get('state') || this.get('state') == "toggle" || this.get('state') == "gold")) {
```
Comment (ghost) [inline tgui/src/components/button.ract]: You don't need this. It didn't work anyways.

File: tgui/src/components/button.styl [hunk 4]
```diff
@@ -39,6 +39,7 @@ span.button
   buttoncolor(selected, button-color-selected)
   buttoncolor(toggle, button-color-selected)
   buttoncolor(caution, button-color-caution)
+  buttoncolor(gold, button-color-gold)
```
Comment (ghost) [inline tgui/src/components/button.styl]: You don't need this. It didn't work anyways.

File: tgui/src/styles/clockwork.styl [hunk 5]
```diff
@@ -32,6 +32,7 @@ body.clockwork
   button-color-disabled     = #2D1400
   button-color-selected     = #CFBA47
   button-color-caution      = yellow-orange
+  button-color-gold         = #FFD700
```
Comment (ghost) [inline tgui/src/styles/clockwork.styl]: You don't need this. It didn't work anyways.

File: tgui/src/styles/syndicate.styl [hunk 6]
```diff
@@ -18,6 +18,7 @@ body.syndicate
   button-color-disabled    = gray
   button-color-selected    = dark-red
   button-color-caution     = yellow-orange
+  button-color-gold        = #FFD700
```
Comment (ghost) [inline tgui/src/styles/syndicate.styl]: You don't need this. It didn't work anyways.

File: tgui/src/tgui.styl [hunk 7]
```diff
@@ -75,6 +75,7 @@ button-color-normal        = royal-blue
 button-color-disabled      = light-gray
 button-color-selected      = green
 button-color-caution       = yellow
+button-color-gold          = #FFD700
```
Comment (ghost) [inline tgui/src/tgui.styl]: You don't need this. It didn't work anyways.

File: yogstation/code/modules/guardian/abilities/special/pocket.dm (hunk 2 above)
Comment (SomeguyManperson) [reply yogstation/code/modules/guardian/abilities/special/pocket.dm:117]: ported the healing in godmode bit

**Pull request merged**